systemtest/benchtest/expvar/metrics_test.go

```diff
@@ -137,16 +133,55 @@ func TestWatchMetricNonBlocking(t *testing.T) {
 	}
 }
 
-func getTestServer(t *testing.T) *httptest.Server {
-	var count uint64
+// setupTestServer creates a test http server with following properties:
+// - max number of responses are controlled using channels
+// - max number of responses is configured using count which represents
+//   the number of samples that the collector witnesses
+// - if the responses are not consumed in 1 second the server returns error
+// - each metric creates a timeseries of natural numbers
+//
+// TODO: @lahsivjar the logic of querying expvar multiple times leaks into
+// these test cases due to the logic in test server to keep the aggregation
+// deterministic for unit testing
+func setupTestServer(t *testing.T, count int, done chan<- bool) *httptest.Server {
+	resChan := make(chan string)
+	timeout := time.Second
+	server := getTestServer(t, resChan, timeout)
+	factorBasedOnQueryExpvar := 12
+
+	go func() {
+		var val int
+		for i := 0; i < count; i++ {
+			val++
+			for j := 0; j < factorBasedOnQueryExpvar; j++ {
+				select {
+				case resChan <- fmt.Sprintf(`{"beat.runtime.goroutines": %d}`, val):
+				case <-time.After(timeout):
+					done <- false
+					return
+				}
+			}
+		}
+		done <- true
+	}()
+
+	return server
+}
+
+func getTestServer(t *testing.T, resChan <-chan string, timeout time.Duration) *httptest.Server {
 	return httptest.NewServer(
```

Comment:
Should we be closing this server with a t.Cleanup?

Reply:
Sounds good, will update the PR